BLD: Update vendor-meson to fix module_feature conflicts arguments bug

seiko2plus opened the pull request:
Bug detected and fixed by @Vogtinator in PR numpy/meson#12. The issue was located in the Meson features module, and implementing the fix will require updating the vendored Meson repository.

seiko2plus commented:
Fixed the build but I got segfault, maybe its qemu bug. re-test it after brings native armhf support via #28653

rgommers commented:
> @rgommers Is the meson update OK?

Yes that's on purpose here. Once CI is green here, https://github.com/numpy/meson/pull/12 can be merged and this PR updated to include the `main-numpymeson` branch of https://github.com/numpy/meson

charris commented:
The failure looks meson related. Caching? Or maybe the submodule needs an update.

```
    raise RuntimeError(
        "Meson configuration failed; please try `spin build` again with the `--clean` flag."
    )
```

seiko2plus commented:
> The failure looks meson related. Caching? Or maybe the submodule needs an update.

That's true, the submodule needs a rebase since it's known blas issue this has been fixed in later meson versions, but it can be ignored. What matters is the armhf error needs to be green as Ralf explained.


<details>
<summary>current armhf error:</summary>
numpy/_core/tests/test_memmap.py .                                       [  0%]
numpy/_core/tests/test_multiarray.py ................................... [  0%]
....................................................s................... [  0%]
............................................XX.......................... [  0%]
.Fatal Python error: Aborted

Current thread 0x41007020 (most recent call first):
  File "/numpy/build-install/usr/lib/python3/dist-packages/numpy/_core/fromnumeric.py", line 1122 in sort
  File "/numpy/build-install/usr/lib/python3/dist-packages/numpy/_core/tests/test_multiarray.py", line 2094 in test_sort
  File "/usr/local/lib/python3.10/dist-packages/_pytest/python.py", line 159 in pytest_pyfunc_call
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/local/lib/python3.10/dist-packages/_pytest/python.py", line 1627 in runtest
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line 174 in pytest_runtest_call
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line 242 in <lambda>
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line 341 in from_call
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line 241 in call_and_report
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line [132](https://github.com/numpy/numpy/actions/runs/14251604055/job/39945473229?pr=28635#step:10:133) in runtestprotocol
  File "/usr/local/lib/python3.10/dist-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/local/lib/python3.10/dist-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/local/lib/python3.10/dist-packages/_pytest/main.py", line 337 in _main
  File "/usr/local/lib/python3.10/dist-packages/_pytest/main.py", line 283 in wrap_session
  File "/usr/local/lib/python3.10/dist-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_callers.py", line 103 in _multicall
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/usr/local/lib/python3.10/dist-packages/pluggy/_hooks.py", line 513 in __call__
  File "/usr/local/lib/python3.10/dist-packages/_pytest/config/__init__.py", line 175 in main
  File "/usr/local/lib/python3.10/dist-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/usr/local/bin/pytest", line 8 in <module>

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, numpy._core._multiarray_tests, numpy._core._rational_tests, numpy._core._umath_tests, cython.cimports.libc.math, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, numpy._core._struct_ufunc_tests, numpy._core._simd, numpy._core._operand_flag_tests, numpy.linalg.lapack_lite (total: 19)
</details>

close/open,  re-test we have native armhf support now.

seiko2plus commented:
@rgommers, I moved link error fix into a separated pull-request #28671

seiko2plus committed:
Update vendor-meson to fix module_feature conflicts arguments bug

Fix incorrect handling of the compiler arguments 'conflicts' that
was causing failures when detecting asmid instructions at compile time on
aarch32 platforms.

charris commented:
Thanks Sayed.
